.github/workflows/enhancement.yml

```diff
@@ -0,0 +1,18 @@
+# this workflow runs only when the 'enhancement' label is added, and not subsequently
+name: enhancement-label
+on: 
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+    types: [ labeled ]
+jobs:
+  enhancement-specific-stuff:
+    if: ${{ github.event.label.name == 'enhancement' }}
```

Comment:
Yes, that is correct - they are incompatible. This particular workflow would only run the moment a PR is labeled.

Removed the "on push" trigger to better reflect the use case (as it was misleading anyway).

Reply:
I just tested, and it works with `github.event.label...`
I reckon it's because `pull_request` is already a type of `event`.
A similar thing is in the official documentation, but for issues:
https://docs.github.com/en/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added